RELEASE.md

```diff
@@ -34,4 +36,39 @@ resolvers ++= Seq(
   )
 ```
 
-## TODO(tomerk): Add details about publishing a full release
+## Full Release
+
+1. Update the build.sbt and the Makefile to the new version
+
+2. Cut a release branch for major version releases
```

Comment:
I can be explicit about the next step (tagging the commit). This step is much more variable depending on what exact commits we want to include (whether we want to do any commit cherry-picking, etc.)